lib/pluginmanager/unpack.rb

```diff
@@ -0,0 +1,47 @@
+require "logstash/util/compress"
+require "fileutils"
+
+class LogStash::PluginManager::Unpack < LogStash::PluginManager::Command
+
+  option "--tgz", :flag, "unpack a packaged tar.gz file", :default => !LogStash::Environment.windows?
+  option "--zip", :flag, "unpack a packaged  zip file", :default => LogStash::Environment.windows?
+  option "--force", :flag, "Override any previously generated bundle", :default => false
+
+  parameter "file", "the package file name", :attribute_name => :package_file, :required => true
+
+  def execute
+    puts("Unpacking #{package_file} to be used as a plugins local library")
+
+    FileUtils.rm_rf(LogStash::Environment::CACHE_PATH) if force?
+    validate_cache_location
+    unpack(package_file)
+
+    puts("Unpacked at #{LogStash::Environment::CACHE_PATH}")
+    puts("If you want to use the generated library to install/update plugins you should call bin/plugin with the --local flag")
```

Comment:
"the unpacked plugins can now be installed in local-only mode using bin/plugin with the --local flag"